src/charm.py

```diff
@@ -0,0 +1,469 @@
+        return pid
+
+    def set_topic(self, topic_name: str) -> str:
+        """Set topic on the relation databag."""
+        if relation_data := self.charm.model.get_relation(self.name):
+            self.app_data.copy(update={"topic_name": topic_name}).write(
+                relation_data.data[self.charm.app]
+            )
+        return topic_name
+
+    def set_database(self, database_name: str) -> str:
+        """Set database name in the peer relation databag."""
+        if relation_data := self.charm.model.get_relation(self.name):
+            self.app_data.copy(update={"database_name": database_name}).write(
+                relation_data.data[self.charm.app]
+            )
+        return database_name
+
+    def set_private_key(self, private_key: str) -> str:
+        """Set private key in the peer relationd databag."""
+        if relation_data := self.charm.model.get_relation(self.name):
+            self.app_data.copy(update={"private_key": private_key}).write(
+                relation_data.data[self.charm.app]
+            )
+        return private_key
+
+    @property
+    def unit_data(self) -> PeerRelationUnitData:
+        """Return the unit databag."""
+        parsed = (
+            get_relation_data_as(PeerRelationUnitData, relation.data[self.charm.unit])
+            if (relation := self.charm.model.get_relation(self.name))
+            else None
+        )
+
+        if isinstance(parsed, ValidationError):
+            logger.error(f"There was a problem to read {self.name} databag: {parsed}")
+
+        return parsed if isinstance(parsed, PeerRelationUnitData) else PeerRelationUnitData()
+
+    @property
+    def app_data(self) -> PeerRelationAppData:
+        """Return the peer relation databag."""
+        parsed = (
+            get_relation_data_as(PeerRelationAppData, relation.data[self.charm.app])
+            if (relation := self.charm.model.get_relation(self.name))
+            else None
+        )
+
+        if isinstance(parsed, ValidationError):
+            logger.error(f"There was a problem to read {self.name} databag: {parsed}")
+
+        return (
+            parsed
+            if isinstance(parsed, PeerRelationAppData)
+            else PeerRelationAppData(topic_name=self.charm.config.topic_name)
+        )
+
+
+class KafkaAppCharm(TypedCharmBase[CharmConfig]):
+    """Charm the service."""
+
+    config_type = CharmConfig
+
+    def __init__(self, *args):
+        super().__init__(*args)
+
+        self.framework.observe(self.on.install, self._on_install)
+        self.framework.observe(self.on.config_changed, self._on_config_changed)
+
+        self.peer_relation = PeerRelation(self, name=PEER)
+        self.certificates = TLSCertificatesRequiresV1(self, CERTIFICATES)
+
+        self.kafka_cluster = KafkaRequires(
+            self,
+            relation_name=KAFKA_CLUSTER,
+            topic=self.config.topic_name,
+            extra_user_roles=self.config.app_type.value,
```

Comment:
**issue** shouldn't specify the consumer group prefix here?

Reply:
Good catch, I was using an old version of the `data_interfaces` lib